### DISCUSSION TITLE
NameError: name 'fdownloadmodel' is not defined

### DISCUSSION THREAD

**Discussion by SlightlyMe** [TheLastBen/fast-stable-diffusion#578] [Q&A]
Hi, I'm running Dreambooth for the first time using colab, and when running the training cell, after uploading the images I get the following error. Pretty sure it's a newby mistake, but, any help is a appreciated. Thanks

Original model not found, downloading....
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-4-914e56deea7a> in <module>
     15     get_ipython().system("rm -r '/content/stable-diffusion-v1-5'")
     16   print('Original model not found, downloading....')
---> 17   fdownloadmodel()
     18   if os.path.exists('/content/stable-diffusion-v1-5/unet/diffusion_pytorch_model.bin'):
     19      print('Model downloaded, proceeding to training...')

NameError: name 'fdownloadmodel' is not defined

**Comment by TheLastBen** [comment]
Hi, make sure you correctly run the "Load/Create Session"

**Reply by SlightlyMe** [reply]
Thanks. In the end I started again, and again... I seem to get errors at different stages, but perseverance will win out :D

**Reply by TheLastBen** [reply]
Run all the cells one by one and all should run smoothly

**Reply by SlightlyMe** [reply]
yeah, I've just had a collection of little hic-ups, including colab telling me I've out of gpu juice 50% into training, and after buying the credits had to restart again!
In the end I deleted the instance and started again, 30% in as I type :D

**Reply by SlightlyMe** [reply]
runtime stopped at 92%!!!

when I restarted it I get the below error. Can I continue this without losing progress?
Original model not found, downloading....
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
[<ipython-input-2-3e8f4f69f22c>](https://localhost:8080/#) in <module>
     15     get_ipython().system("rm -r '/content/stable-diffusion-v1-5'")
     16   print('Original model not found, downloading....')
---> 17   fdownloadmodel()
     18   if os.path.exists('/content/stable-diffusion-v1-5/unet/diffusion_pytorch_model.bin'):
     19      print('Model downloaded, proceeding to training...')

NameError: name 'fdownloadmodel' is not defined

**Reply by SlightlyMe** [reply]
don't no where the bold came from!

**Reply by SlightlyMe** [reply]
I tried load/create session but it gives the following error:
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
[<ipython-input-3-297fa109a230>](https://localhost:8080/#) in <module>
      2 from IPython.display import clear_output
      3 from IPython.utils import capture
----> 4 import wget
      5 import time
      6 

ModuleNotFoundError: No module named 'wget'

---------------------------------------------------------------------------
NOTE: If your import is failing due to a missing package, you can
manually install dependencies using either !pip or !apt.

To view examples of installing some common dependencies, click the
"Open Examples" button below.
---------------------------------------------------------------------------

**Reply by SlightlyMe** [reply]
really not having much luck!
now, even with a fresh instance, carefully going through all the cells, I reach training and gt the following error.

Progress:|                         |  0% 1/6200 [00:14<24:33:16, 14.26s/it, loss=0.131, lr=2e-6] MyPics Traceback (most recent call last):
  File "/content/diffusers/examples/dreambooth/train_dreambooth.py", line 788, in <module>
    main()
  File "/content/diffusers/examples/dreambooth/train_dreambooth.py", line 657, in main
    noise_pred = unet(noisy_latents, timesteps, encoder_hidden_states).sample
  File "/usr/local/lib/python3.7/dist-packages/torch/nn/modules/module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/accelerate/utils/operations.py", line 507, in __call__
    return convert_to_fp32(self.model_forward(*args, **kwargs))
  File "/usr/local/lib/python3.7/dist-packages/torch/amp/autocast_mode.py", line 12, in decorate_autocast
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/diffusers/models/unet_2d_condition.py", line 314, in forward
    upsample_size=upsample_size,
  File "/usr/local/lib/python3.7/dist-packages/torch/nn/modules/module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/diffusers/models/unet_blocks.py", line 1150, in forward
    hidden_states = resnet(hidden_states, temb)
  File "/usr/local/lib/python3.7/dist-packages/torch/nn/modules/module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/diffusers/models/resnet.py", line 359, in forward
    hidden_states = self.norm1(hidden_states)
  File "/usr/local/lib/python3.7/dist-packages/torch/nn/modules/module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/torch/nn/modules/normalization.py", line 273, in forward
    input, self.num_groups, self.weight, self.bias, self.eps)
  File "/usr/local/lib/python3.7/dist-packages/torch/nn/functional.py", line 2516, in group_norm
    return torch.group_norm(input, num_groups, weight, bias, eps, torch.backends.cudnn.enabled)
RuntimeError: CUDA out of memory. Tried to allocate 30.00 MiB (GPU 0; 14.76 GiB total capacity; 13.36 GiB already allocated; 11.75 MiB free; 13.41 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF
Progress:|                         |  0% 1/6200 [00:15<26:19:51, 15.29s/it, loss=0.131, lr=2e-6]
Traceback (most recent call last):
  File "/usr/local/bin/accelerate", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.7/dist-packages/accelerate/commands/accelerate_cli.py", line 43, in main
    args.func(args)
  File "/usr/local/lib/python3.7/dist-packages/accelerate/commands/launch.py", line 837, in launch_command
    simple_launcher(args)
  File "/usr/local/lib/python3.7/dist-packages/accelerate/commands/launch.py", line 354, in simple_launcher
    raise subprocess.CalledProcessError(returncode=process.returncode, cmd=cmd)
subprocess.CalledProcessError: Command '['/usr/bin/python3', '/content/diffusers/examples/dreambooth/train_dreambooth.py', '--image_captions_filename', '--train_text_encoder', '--save_starting_step=500', '--stop_text_encoder_training=6200', '--save_n_steps=0', '--Session_dir=/content/gdrive/MyDrive/Fast-Dreambooth/Sessions/MyModel4', '--pretrained_model_name_or_path=/content/stable-diffusion-v1-5', '--instance_data_dir=/content/gdrive/MyDrive/Fast-Dreambooth/Sessions/MyModel4/instance_images', '--output_dir=/content/models/MyModel4', '--instance_prompt=', '--seed=822907', '--resolution=1024', '--mixed_precision=fp16', '--train_batch_size=1', '--gradient_accumulation_steps=1', '--use_8bit_adam', '--learning_rate=2e-6', '--lr_scheduler=polynomial', '--lr_warmup_steps=0', '--max_train_steps=6200']' returned non-zero exit status 1.
Something went wrong

**Reply by SlightlyMe** [reply]
Now new instances I can't get past create/load session without this error
mkdir: cannot create directory ‘/content/gdrive/MyDrive’: Transport endpoint is not connected

really not working for me

**Reply by TheLastBen** [reply]
If you face any issue, disconnect from the colab runtime an reconnect, then rerun all the cells again.

The error you're getting at the beginning of the training is caused by VRAM, make sure you use a resolution below 640
